src/lib/updateGrantsRecipients.js

```diff
@@ -185,16 +185,30 @@ export const updateCDIGrantsWithOldGrantData = async (grantsToUpdate) => {
     const updates = grantsToUpdate.map(async (grant) => {
       // eslint-disable-next-line max-len
       const replacedGrants = await GrantReplacements.findAll({ where: { replacingGrantId: grant.id } });
+
+      // If we don't have any replaced grants replacements we have nothing to do for this grant.
+      // Prevent confusion of throwing exception below.
+      if (!replacedGrants.length) {
+        logger.info(`updateCDIGrantsWithOldGrantData: No grant replacements found for CDI grant: ${grant.id}, skipping`);
+        return Promise.resolve();
+      }
+
       // eslint-disable-next-line max-len
       const validOldGrants = (await Promise.all(replacedGrants.map((rg) => Grant.findByPk(rg.replacedGrantId)))).filter(Boolean);
 
       const [regionId] = uniq(validOldGrants.map((g) => g.regionId));
       const [recipientId] = uniq(validOldGrants.map((g) => g.recipientId));
 
-      if (!regionId || !recipientId || validOldGrants.length !== replacedGrants.length) {
+      if (!regionId || !recipientId) {
         throw new Error(`Expected one region and recipient for grant ${grant.id}, got ${validOldGrants.length} valid grants`);
       }
 
+      // Ensure allValidOldGrants have the same recipient and region.
+      if (!validOldGrants.every((g) => g.regionId === regionId && g.recipientId === recipientId)) {
+        throw new Error(`Expected all valid replaced grants to have the same recipient and region for CDI grant ${grant.id}`);
```

Comment:
do we want to throw here instead of just logging and moving on to the next grant in the map? here's why I ask:

if `grantsToUpdate` contains 10 grants, and we throw on the 7th grant, that leaves 3 remaining grants that never get processed, right?

this comment applies also to the thrown error on 203

Reply:
So your suggesting adding another try catch inside the actual processing of each grant. This way we continue processing the remaining grants, right?

Reply:
yep or just don't use promise.all - however you'd like to approach it

Reply:
Updated to log message and skip.